syft/serde/protobuf/serde.py

```diff
@@ -0,0 +1,356 @@
+            and deserialization part, but we don't need to unbufferize all the message.
+            This is the case for Plan workers for instance
+
+    Returns:
+        object: the deserialized form of the binary input.
+    """
+    if worker is None:
+        # TODO[jvmancuso]: This might be worth a standalone function.
+        worker = syft.framework.hook.local_worker
+
+    # 1) Decompress the binary if needed
+    binary = compression._decompress(binary)
+
+    # 2) Deserialize
+    msg_wrapper = SyftMessagePB()
+    msg_wrapper.ParseFromString(binary)
+
+    # 3) Convert back to a Python object
+    message_type = msg_wrapper.WhichOneof("contents")
+    python_obj = _unbufferize(worker, getattr(msg_wrapper, message_type))
+    return python_obj
+
+
+def create_protobuf_id(id) -> IdPB:
+    protobuf_id = IdPB()
+    if type(id) == type("str"):
+        protobuf_id.id_str = id
+    else:
+        protobuf_id.id_int = id
+    return protobuf_id
+
+
+def _bufferize(worker: AbstractWorker, obj: object, **kwargs) -> object:
+    """
+    This function takes an object as input and returns a
+    Protobuf object. The reason we have this function
+    is that some objects are either NOT supported by high level (fast)
+    serializers OR the high level serializers don't support the fastest
+    form of serialization. For example, PyTorch tensors have custom pickle
+    functionality thus its better to pre-serialize PyTorch tensors using
+    pickle and then serialize the binary in with the rest of the message
+    being sent.
+
+    Args:
+        obj: An object which needs to be converted to Protobuf.
+
+    Returns:
+        An Protobuf object which Protobuf can serialize.
+    """
+
+    # Check to see if there is a bufferizer
+    # for this type. If there is, return the bufferized object.
+    # breakpoint()
+    current_type = type(obj)
+    if current_type in bufferizers:
+        result = bufferizers[current_type](worker, obj, **kwargs)
+        return result
+
+    # If we already tried to find a bufferizer for this type but failed, we should
+    # just return the object as it is.
+    elif current_type in no_bufferizers_found:
+        raise Exception(f"No corresponding Protobuf message found for {current_type}")
+
+    else:
+        # If the object type is not in bufferizers,
+        # we check the classes that this object inherits from.
+        # `inspect.getmro` give us all types this object inherits
+        # from, including `type(obj)`. We can skip the type of the
+        # object because we already tried this in the
+        # previous step.
+        classes_inheritance = inspect.getmro(type(obj))[1:]
+
+        for inheritance_type in classes_inheritance:
+            if inheritance_type in bufferizers:
+                # Store the inheritance_type in bufferizers so next time we see this type
+                # serde will be faster.
+                bufferizers[current_type] = bufferizers[inheritance_type]
```

Comment:
Should we also have `inherited_simplifiers_found` here?

Reply:
Yes! Good catch.